clientv3/balancer.go

```diff
@@ -131,6 +147,59 @@ func (b *simpleBalancer) updateAddrs(eps []string) {
 	}
 }
 
+func (b *simpleBalancer) hasPinAddr(addrs []grpc.Address) bool {
+	for _, addr := range addrs {
+		if addr.Addr == b.pinAddr {
+			return true
+		}
+	}
+	return false
+}
+
+func (b *simpleBalancer) sendLoop() {
+	defer close(b.stopc)
+	var (
+		upc   chan struct{}
```

Comment:
this is too widely scoped; it's reinitialized at the beginning of the for loop; have `upc := ` under the rlock instead